src/Grenade/Core/Layer.hs

```diff
@@ -85,3 +90,17 @@ class UpdateLayer x => Layer x (i :: Shape) (o :: Shape) where
   --   Returns the gradient layer and the derivatives to push back
   --   further.
   runBackwards   :: x -> Tape x i o -> S o -> (Gradient x, S i)
+
+
+-- | Class for random initialization of a layer. This enables to use
+--   various initialization techniques for the networks. Every layer
+--   needs to implement this.
+class RandomLayer x where
```

Comment:
Does pulling this into a separate class buy us anything?

Maybe UpdateLayer should just be renamed to BaseLayer or something and hold both?

Reply:
Yes, I wanted to keep `UpdateLayer` for backward compatibility. But renaming is fine as well. I am currently on vacation and look into the other issues soon.